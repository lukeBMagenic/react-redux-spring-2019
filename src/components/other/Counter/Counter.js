import React, { Component } from 'react';
import './Counter.scss';

class Counter extends Component {
  constructor(props) {
    super(props);
    this.state = {count: 0};
  }

  increment = () => {
    this.setState({
      count: this.state.count + 1
    })
  }
  
  decrement = () => {
    this.setState({
      count: this.state.count - 1
    })
  }

  render() {
    return (
      <div>
        <h1>
          <button onClick={this.increment}>Increment</button>
          <button onClick={this.decrement}>Decrement</button>
        </h1>
        <h1>{this.state.count}</h1>
      </div>
    );
  }
}

export default Counter;
