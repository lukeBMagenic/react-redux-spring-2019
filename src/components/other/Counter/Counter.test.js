import React from 'react';
import { shallow } from "enzyme";

import Counter from './Counter';

describe("Counter", () => {
  it("should render correctly", () => {
    const component = shallow(<Counter />);

    expect(component).toMatchSnapshot();
  });
});
