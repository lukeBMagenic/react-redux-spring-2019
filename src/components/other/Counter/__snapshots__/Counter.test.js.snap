// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly 1`] = `
<div>
  <h1>
    <button
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      onClick={[Function]}
    >
      Decrement
    </button>
  </h1>
  <h1>
    0
  </h1>
</div>
`;
